instrumentation/google.golang.org/grpc/otelgrpc/test/interceptor_test.go

```diff
@@ -583,39 +583,150 @@ func TestStreamClientInterceptorWithError(t *testing.T) {
 	assert.Equal(t, codes.Error, span.Status().Code)
 }
 
-func TestServerInterceptorError(t *testing.T) {
+var serverChecks = []struct {
+	grpcCode grpc_codes.Code
+	spanCode codes.Code
+	name     string
+}{
+	{
+		grpcCode: grpc_codes.OK,
+		spanCode: codes.Unset,
+		name:     "ok",
+	},
+	{
+		grpcCode: grpc_codes.Canceled,
+		spanCode: codes.Unset,
+		name:     "cancelled",
+	},
+	{
+		grpcCode: grpc_codes.NotFound,
+		spanCode: codes.Unset,
+		name:     "not.found",
+	},
+	{
+		grpcCode: grpc_codes.PermissionDenied,
+		spanCode: codes.Unset,
+		name:     "permission.denied",
+	},
+	{
+		grpcCode: grpc_codes.Unimplemented,
+		spanCode: codes.Error,
+		name:     "unimplemented",
+	},
+	{
+		grpcCode: grpc_codes.Internal,
+		spanCode: codes.Error,
+		name:     "internal",
+	},
+	{
+		grpcCode: grpc_codes.Unauthenticated,
+		spanCode: codes.Unset,
+		name:     "unauthenticated",
+	},
+}
+
+func UnaryServerInterceptor(t *testing.T) {
 	sr := tracetest.NewSpanRecorder()
 	tp := trace.NewTracerProvider(trace.WithSpanProcessor(sr))
 	usi := otelgrpc.UnaryServerInterceptor(otelgrpc.WithTracerProvider(tp))
-	deniedErr := status.Error(grpc_codes.PermissionDenied, "PERMISSION_DENIED_TEXT")
-	handler := func(_ context.Context, _ interface{}) (interface{}, error) {
-		return nil, deniedErr
-	}
-	_, err := usi(context.Background(), &grpc_testing.SimpleRequest{}, &grpc.UnaryServerInfo{}, handler)
-	require.Error(t, err)
-	assert.Equal(t, err, deniedErr)
+	for _, check := range serverChecks {
```

Comment:
How about adding a `t.Run(check.grpcCode.String(), func ...`? 
Then the `require` as far as I remember should not break the whole "test suite" and we could use `require` instead of `if !assert... { continue}` in some places.

PS. Same comment for the second test.